src/api/user/ActiveRoleSynchronizer.js

```diff
@@ -0,0 +1,41 @@
+import { ACTIVE_ROLE_BROADCAST_CHANNEL_NAME } from './constants';
+
+class ActiveRoleSynchronizer {
+  #roleChannel;
+
+  constructor(openmct) {
+    this.openmct = openmct;
+    this.#roleChannel = new BroadcastChannel(ACTIVE_ROLE_BROADCAST_CHANNEL_NAME);
+    this.setActiveRoleFromChannelMessage = this.setActiveRoleFromChannelMessage.bind(this);
+
+    this.subscribeToRoleChanges(this.setActiveRoleFromChannelMessage);
+  }
+  extractRoleFromEvent(callback) {
+    return function (event) {
+      callback(event.data);
+    };
+  }
+  subscribeToRoleChanges(callback) {
+    this.#roleChannel.addEventListener('message', this.extractRoleFromEvent(callback));
+  }
+  unsubscribeFromRoleChanges(callback) {
+    this.#roleChannel.removeEventListener('message', this.extractRoleFromEvent(callback));
```

Comment:
@akhenry - I've removed the extractRoleFromEvent as suggested. The original intent of `extractRoleFromEvent` was to prevent the leaking of the Broadcast Channel event from leaking into the UserIndicator since it also subscribes to role changes. I've added a method in UserIndicator called `onRoleChange` to unpack the event.data and pass the role into `setRoleSelection`.